dockers-install.sh

```diff
@@ -0,0 +1,39 @@
+# Update
+sudo apt update
+
+# Installing ca-certificates
+if (which ca-certificates)
+then
+  echo "ca-certificates already installed"
+else
+  echo "Installing ca-certificates"
+  sudo apt install ca-certificates
+fi
+
+# Installing curl
+if (which curl)
+then
+  echo "curl already installed"
+else
+  echo "Installing curl"
+  sudo apt install curl
+fi
+
+# Creating a keyrings directory
+if (test -f /etc/bin/keyrings)
+then
+  echo "creating keyrings"
+else
+  echo "Creating keyrings"
+  sudo install -m 0755 -d /etc/apt/keyrings
+fi
+
+# Adding gpg keyring, and configuring
+if (which gpg)
```

Comment:
The check here would be checking the saved file not if gpg is installed like this:

`if (gpg /etc/apt/keyrings/docker.asc)`

Also you have 2 commands here, one that writes the key and another that changes it's mode using `chmod`. Maybe break out the chmod command and try to write a test for it?

Reply:
I did have it like that, and put the command on the end of the install block to see if it worked that way.
```if (test -f /etc/bin/keyrings/docker.asc)
then
  echo " gpg keyring already configured"
else
  echo "configuring gpg keyrings"
  sudo chmod a+r /etc/apt/keyrings/docker.asc
fi```